File: scripts/addRecipe.py
#!/usr/bin/env python3

"""
Script para adicionar novas receitas ao arquivo recipes.js

Uso: python3 scripts/addRecipe.py

O script irá:
1. Ler o arquivo recipes.js
2. Solicitar os dados da nova receita
3. Adicionar a receita com o próximo ID disponível
4. Salvar o arquivo atualizado
"""

import re
import os
import sys
from pathlib import Path

def find_last_id(content):
    """Encontra o último ID no arquivo"""
    ids = re.findall(r'id: (\d+)', content)
    return max(int(id) for id in ids) if ids else 0

def format_ingredient(quantity, item):
    """Formata um ingrediente"""
    return f'      {{ quantity: "{quantity}", item: "{item}" }}'

def format_recipe(recipe_id, name, ingredients):
    """Formata uma receita completa"""
    ingredients_str = ',\n'.join(
        format_ingredient(qty, item) for qty, item in ingredients
    )
    
    return f"""  {{
    id: {recipe_id},
    name: "{name}",
    ingredients: [
{ingredients_str}
    ]
  }}"""

def main():
    try:
        # Caminho do arquivo
        script_dir = Path(__file__).parent
        recipes_path = script_dir.parent / 'src' / 'data' / 'recipes.js'
        
        if not recipes_path.exists():
            print(f"❌ Arquivo não encontrado: {recipes_path}")
            sys.exit(1)
        
        # Ler o arquivo
        with open(recipes_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Encontrar o próximo ID
        last_id = find_last_id(content)
        next_id = last_id + 1
        
        print('\n🍕 ADICIONAR NOVA RECEITA\n')
        print(f'Próximo ID: {next_id}\n')
        
        # Solicitar dados
        name = input('Nome da receita: ').strip()
        
        if not name:
            print('❌ Nome não pode estar vazio!')
            sys.exit(1)
        
        print('\nIngredientes (digite "pronto" quando terminar):')
        ingredients = []
        ingredient_count = 1
        
        while True:
            quantity = input(f'Ingrediente {ingredient_count} - Quantidade (ex: 1x, 200gr, 1 pacote): ').strip()
            
            if quantity.lower() == 'pronto':
                break
            
            if not quantity:
                print('⚠️  Quantidade não pode estar vazia!')
                continue
            
            item = input(f'Ingrediente {ingredient_count} - Nome: ').strip()
            
            if not item:
                print('⚠️  Nome do ingrediente não pode estar vazio!')
                continue
            
            ingredients.append((quantity, item))
            ingredient_count += 1
        
        if not ingredients:
            print('❌ Nenhum ingrediente foi adicionado!')
            sys.exit(1)
        
        # Formatar a receita
        recipe_str = format_recipe(next_id, name, ingredients)
        
        # Adicionar ao arquivo
        updated_content = content.replace(
            '\n];',
            f',\n{recipe_str}\n];'
        )
        
        # Salvar o arquivo
        with open(recipes_path, 'w', encoding='utf-8') as f:
            f.write(updated_content)
        
        # Mostrar resultado
        print(f'\n✅ Receita adicionada com sucesso!')
        print(f'📋 ID: {next_id}')
        print(f'📝 Nome: {name}')
        print(f'🥘 Ingredientes: {len(ingredients)}')
        print('\n💾 Arquivo salvo! Agora execute:')
        print('   git add .')
        print(f'   git commit -m "Adicionar receita: {name}"')
        print('   git push')
        
    except Exception as e:
        print(f'❌ Erro: {str(e)}')
        sys.exit(1)

if __name__ == '__main__':
    main()

File: scripts/addRecipe.js
#!/usr/bin/env node

/**
 * Script para adicionar novas receitas ao arquivo recipes.js
 * 
 * Uso: node scripts/addRecipe.js
 * 
 * O script irá:
 * 1. Ler o arquivo recipes.js
 * 2. Solicitar os dados da nova receita
 * 3. Adicionar a receita com o próximo ID disponível
 * 4. Salvar o arquivo atualizado
 */

import fs from 'fs';
import path from 'path';
import readline from 'readline';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const question = (prompt) => new Promise((resolve) => {
  rl.question(prompt, resolve);
});

async function main() {
  try {
    const recipesPath = path.join(__dirname, '../src/data/recipes.js');
    
    // Ler o arquivo atual
    let content = fs.readFileSync(recipesPath, 'utf8');
    
    // Encontrar o último ID
    const idMatches = content.match(/id: (\d+)/g);
    const lastId = Math.max(...idMatches.map(m => parseInt(m.match(/\d+/)[0])));
    const nextId = lastId + 1;
    
    console.log('\n🍕 ADICIONAR NOVA RECEITA\n');
    console.log(`Próximo ID: ${nextId}\n`);
    
    // Solicitar dados da receita
    const name = await question('Nome da receita: ');
    
    console.log('\nIngredientes (digite "pronto" quando terminar):');
    const ingredients = [];
    let ingredientCount = 1;
    
    while (true) {
      const quantity = await question(`Ingrediente ${ingredientCount} - Quantidade (ex: 1x, 200gr, 1 pacote): `);
      
      if (quantity.toLowerCase() === 'pronto') {
        break;
      }
      
      const item = await question(`Ingrediente ${ingredientCount} - Nome: `);
      
      ingredients.push({ quantity, item });
      ingredientCount++;
    }
    
    if (ingredients.length === 0) {
      console.log('❌ Nenhum ingrediente foi adicionado!');
      rl.close();
      return;
    }
    
    // Criar objeto da receita
    const newRecipe = {
      id: nextId,
      name,
      ingredients
    };
    
    // Formatar a receita como string
    const recipeString = `  {
    id: ${newRecipe.id},
    name: "${newRecipe.name}",
    ingredients: [
${newRecipe.ingredients.map(ing => `      { quantity: "${ing.quantity}", item: "${ing.item}" }`).join(',\n')}
    ]
  }`;
    
    // Adicionar a nova receita antes do fechamento do array
    const updatedContent = content.replace(
      /\n\];$/,
      `,\n${recipeString}\n];`
    );
    
    // Salvar o arquivo
    fs.writeFileSync(recipesPath, updatedContent, 'utf8');
    
    console.log(`\n✅ Receita adicionada com sucesso!`);
    console.log(`📋 ID: ${nextId}`);
    console.log(`📝 Nome: ${name}`);
    console.log(`🥘 Ingredientes: ${ingredients.length}`);
    console.log('\n💾 Arquivo salvo! Agora execute:');
    console.log('   git add .');
    console.log('   git commit -m "Adicionar receita: ' + name + '"');
    console.log('   git push');
    
    rl.close();
  } catch (error) {
    console.error('❌ Erro:', error.message);
    rl.close();
    process.exit(1);
  }
}

main();
